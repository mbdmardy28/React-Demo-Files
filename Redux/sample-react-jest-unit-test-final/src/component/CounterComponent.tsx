import React, { useEffect } from "react";
import useCounter from "../hooks/useCounter";

interface Props {
    onCountChange: (count: number) => void
}
const CounterComponent: React.FC<Props> = ({onCountChange}) => {
    const {count, increment} = useCounter();

    useEffect(() => {
        onCountChange(count);
    })

    return (
        <div>
            <p>Count: {count}</p>
            <button onClick={() => increment()}>
                Increment
            </button>
        </div>
    )
}

export default CounterComponent;