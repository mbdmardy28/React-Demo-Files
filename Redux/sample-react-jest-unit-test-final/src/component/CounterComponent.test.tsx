import React from 'react';
import CounterComponent from './CounterComponent';
import { mount, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe("CounterComponent",() => {
    let onCountChange:any;
    let wrapper:any;
    beforeEach(() => {
        onCountChange = jest.fn();
        wrapper = mount(<CounterComponent onCountChange={onCountChange} />);
    })
    it('should renders counter component ', () => {
        console.log(wrapper.debug())
        expect(wrapper).not.toBeNull();
    })

    it('should show my default text',() => {
        expect(wrapper.find('p').text()).toEqual('Count: 0');
    })

    it('should correctly increments the count by 1', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('p').text()).toEqual('Count: 1')
    })

    it('should correctly increments the count by 4', () => {
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        expect(wrapper.find('p').text()).toEqual('Count: 4')
    })

    it('should onCountChange is invoke', () => {
       expect(onCountChange).toBeCalledTimes(1)
       wrapper.find('button').simulate('click');
       expect(onCountChange).toBeCalledTimes(2)
    })

    it('create snapshot test', () => {
        let onCountMock = jest.fn()
        const tree = renderer
        .create(<CounterComponent onCountChange ={onCountMock}>Facebook</CounterComponent>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})


