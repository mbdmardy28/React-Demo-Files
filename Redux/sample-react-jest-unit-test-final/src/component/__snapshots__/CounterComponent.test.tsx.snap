// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent create snapshot test 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
